Sources/StreamChatSwiftUI/ChatChannel/ChatChannelViewModel.swift

```diff
@@ -259,6 +268,30 @@ open class ChatChannelViewModel: ObservableObject, MessagesDataSource {
             channelController.markRead()
         }
     }
+    
+    private func checkHeaderType() {
+        let type: ChannelHeaderType
+        if !reactionsShown && isMessageThread {
+            type = .messageThread
+        } else if !channel.currentlyTypingUsersFiltered(
+            currentUserId: chatClient.currentUserId
+        ).isEmpty {
+            type = .typingIndicator
+        } else {
+            type = .regular
+        }
+        
+        if type != channelHeaderType {
+            channelHeaderType = type
+        } else if type == .typingIndicator {
+            // Toolbar is not updated when new user starts typing.
+            // Therefore, we shortly update the state to regular to trigger an update.
+            channelHeaderType = .regular
+            DispatchQueue.main.asyncAfter(deadline: .now() + 0.1) { [weak self] in
+                self?.channelHeaderType = .typingIndicator
+            }
+        }
+    }
```

Comment:
Logic here is preeeeeetty messy. I would love to see a unit test here because if we ever touch this and need to do changes...all hell will probably break loose.

Might require some refactoring but I think that would be worth it. But would love to hear your opinion here.

Reply:
Did small update on the method. Let me know if you have some additional ideas for improvement. We need to keep the ugly hack at the end though, otherwise the updates are not sent. Regarding tests, this is already tested in the tests for the `channelHeaderType`.